### ISSUE TITLE
Remove indirect dependencies to github.com/dgrijalva/jwt-go

### ISSUE THREAD

**Issue by turkenh** [upbound/universal-crossplane#168]
### What happened?

We need to remove all dependencies to github.com/dgrijalva/jwt-go since it is [no longer maintained](https://github.com/dgrijalva/jwt-go/issues/462) and contains a [security issue](https://github.com/advisories/GHSA-w73w-5m7g-f7qc).  

We are removing all direct dependencies with #166, however, there are still some indirect dependencies that we cannot simply remove yet. See the discussion here: https://github.com/upbound/universal-crossplane/pull/166#discussion_r677313186

In short, this is currently blocked by the echo framework and they claim that their usage is safe regarding the security issue: https://github.com/labstack/echo/pull/1916#issuecomment-878046299. We need to keep an eye there and apply once a solution is available.

### How can we reproduce it?

Check occurrences of github.com/dgrijalva/jwt-go in go.sum file

//cc @ulucinar @negz
